src/android/CameraLauncher.java

```diff
@@ -547,7 +547,7 @@ private String ouputModifiedBitmap(Bitmap bitmap, Uri uri) throws IOException {
         String realPath = FileHelper.getRealPath(uri, this.cordova);
 
         // Get filename from uri
-        String fileName = realPath != null ?
+        String fileName = realPath != null && !realPath.isEmpty() ?
```

Comment:
This is fine, but you may also consider [TextUtils.isEmpty(realPath)] (http://developer.android.com/reference/android/text/TextUtils.html#isEmpty(java.lang.CharSequence)) -- that is the recommended way in Android to check if a string is null or empty.